posthog/api/user.py

```diff
@@ -157,21 +162,35 @@ def to_representation(self, instance: Any) -> Any:
         return super().to_representation(instance)
 
 
-class UserViewSet(mixins.RetrieveModelMixin, mixins.UpdateModelMixin, viewsets.GenericViewSet):
+class UserViewSet(mixins.RetrieveModelMixin, mixins.UpdateModelMixin, mixins.ListModelMixin, viewsets.GenericViewSet):
     serializer_class = UserSerializer
     permission_classes = [
         permissions.IsAuthenticated,
     ]
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = [
+        "is_staff",
+    ]
     queryset = User.objects.none()
     lookup_field = "uuid"
 
     def get_object(self) -> Any:
         lookup_value = self.kwargs[self.lookup_field]
         if lookup_value == "@me":
             return self.request.user
-        raise serializers.ValidationError(
-            "Currently this endpoint only supports retrieving `@me` instance.", code="invalid_parameter",
-        )
+
+        if not self.request.user.is_staff:
+            raise serializers.ValidationError(
+                "Currently this endpoint only supports retrieving `@me` instance or updates by staff users.",
+                code="invalid_parameter",
+            )
```

Comment:
A 403 would be more relevant than a 400 here:
```suggestion
            raise serializers.PermissionDenied(
                "As a non-staff user you're only allowed to access the `@me` user instance."
            )
```